### PR TITLE
Fix Android release

### PR DESCRIPTION
**Pull request by QuintinWillison**
Work in progress but pushing this PR up for visibility.

Until this is resolved, we're blocked in respect of [releasing version 1.2.7](https://github.com/ably/ably-java/pull/693).

### CODE REVIEW

**Comment by QuintinWillison** [comment]
In current state `./gradlew android:assembleRelease` fails locally for me with:

```
> Task :android:verifyReleaseResources FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':android:verifyReleaseResources'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > 1 exception was raised by workers:
     com.android.builder.internal.aapt.v2.Aapt2Exception: Android resource linking failed
     /Users/quintinwillison/.gradle/caches/transforms-2/files-2.1/e74633aea3eab9faecab9ea3de65e0d9/core-1.0.0/res/values/values.xml:57:5-88:25: AAPT: error: resource android:attr/fontStyle not found.

     /Users/quintinwillison/.gradle/caches/transforms-2/files-2.1/e74633aea3eab9faecab9ea3de65e0d9/core-1.0.0/res/values/values.xml:57:5-88:25: AAPT: error: resource android:attr/font not found.

     /Users/quintinwillison/.gradle/caches/transforms-2/files-2.1/e74633aea3eab9faecab9ea3de65e0d9/core-1.0.0/res/values/values.xml:57:5-88:25: AAPT: error: resource android:attr/fontWeight not found.

     /Users/quintinwillison/.gradle/caches/transforms-2/files-2.1/e74633aea3eab9faecab9ea3de65e0d9/core-1.0.0/res/values/values.xml:57:5-88:25: AAPT: error: resource android:attr/fontVariationSettings not found.

     /Users/quintinwillison/.gradle/caches/transforms-2/files-2.1/e74633aea3eab9faecab9ea3de65e0d9/core-1.0.0/res/values/values.xml:57:5-88:25: AAPT: error: resource android:attr/ttcIndex not found.
```

File: android/build.gradle [hunk 1]
```diff
@@ -19,14 +18,12 @@ ext {
...
     repositories {
-        jcenter()
         google()
     }
 }
 
 android {
-    compileSdkVersion 22
-    buildToolsVersion '28.0.3'
+    compileSdkVersion 30
```
Comment (KacperKluka) [inline android/build.gradle:26]: Shouldn't we also update the `targetSdkVersion` to match the `compileSdkVersion`? :thinking:

**Review by paddybyers — APPROVED** [review]
LGTM, thanks

**Review by martin-morek — APPROVED** [review]

File: android/build.gradle (hunk 1 above)
Comment (QuintinWillison) [reply android/build.gradle:26]: No. Not at all. Very different concepts.

**Pull request merged**